keps/sig-storage/177-volume-snapshot/tighten-validation-webhook-crd.md

```diff
@@ -0,0 +1,803 @@
+|  CREATE   |      spec.Source       |                                                                                                                                                                          Exactly one of VolumeHandle (dynamic snapshot created by controller) or SnapshotHandle (pre-provisioned snapshot created by CA) should be specified                                                                                                                                                                          |    400     |
+|  UPDATE   |      spec.Source       |                                                                                                                                                                                                                                    Immutable, no updates allowed.                                                                                                                                                                                                                                     |    400     |
+|  CREATE   | spec.VolumeSnapshotRef | Must have both name and namespace fields set. Preprovisioned: This is the reference to the yet to be created VolumeSnapshot object which should bind to this VolumeSnapshotContent. https://github.com/kubernetes-csi/external-snapshotter/blob/097b1fc7d7cd6576182ca34512c14de1c84b2127/pkg/apis/volumesnapshot/v1beta1/types.go#L270. Dynamic: This is the reference to the VS object which triggered the creation of this VSContent. It also has the UID field, but this is set by the controller. |    400     |
+|  UPDATE   | spec.VolumeSnapshotRef |                                                                                                                                                                                                                                    Immutable, no updates allowed, once it's UID has been set.                                                                                                                                                                                                                                     |    400     |
+
+### Authentication
+
+There are two directions to authentication. Authenticating the webhook server is legitimate, and authenticating the k8s api server is legitimate. 
+
+The API server authenticates the webhook server through TLS certificates and HTTPS. This is required, and an example method of deploying the webhook server with HTTPS will be provided.
```

Comment:
It will be the user's responsibility to handle certificate management and deployment. However a sample script will be provided which can create TLS certificates and upload them to the cluster as a `Secret` object, to be consumed by the webhook server.